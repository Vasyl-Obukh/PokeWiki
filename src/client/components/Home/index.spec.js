import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Home, getCurrentPage, mapStateToProps } from './index';
import configureStore from 'redux-mock-store';

describe('<Home /> tests', () => {
  const mockStore = configureStore();
  const initialState = {cards: {data: {}}};
  const store = mockStore(initialState);
  let props;

  beforeEach(() => {
    props = {
      currentPage: 1,
      showPagination: true,
      searchParams: new URLSearchParams('types=fire,water')
    };
  });

  test('Should render with pagination', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Home {...props} />
        </Router>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render without pagination', () => {
    props.showPagination = false;
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Home {...props} />
        </Router>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly if searchParams prop is not provided', () => {
    delete props.searchParams;

    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Home {...props} />
        </Router>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<Home /> mapStateToProps() tests', () => {
  let props, state;

  beforeEach(() => {
    state = {cards: {data: {count: 27}, isLoading: false}};
    props = {location: {search: 'page=2&&evoLevels=1,3'}};
  });

  test('Show pagination property of the returned object should be truthly', () => {
    expect(mapStateToProps(state, props)).toEqual({
      showPagination: true,
      currentPage: 2,
      searchParams: new URLSearchParams(props.location.search)
    });
  });

  test('Show pagination property of the returned object should be falsy while loading', () => {
    state.cards.isLoading = true;
    expect(mapStateToProps(state, props)).toEqual({
      showPagination: false,
      currentPage: 2,
      searchParams: new URLSearchParams(props.location.search)
    });
  });

  test('Show pagination property of the returned object should be false if there is less than 2 pages', () => {
    state.cards.data.count = 0;
    expect(mapStateToProps(state, props)).toEqual({
      showPagination: false,
      currentPage: 2,
      searchParams: new URLSearchParams(props.location.search)
    });
  });
});

describe('<Home /> helper functions tests', () => {
  test('getCurrentPage() should return page number that is passed to search string', () => {
    const searchParams = new URLSearchParams('?page=7');
    expect(getCurrentPage(searchParams)).toBe(7);
  });

  test('getCurrentPage() should return page number that is typecasted to integer from value passed to search string', () => {
    const searchParams = new URLSearchParams('?page=14,&fafs');
    expect(getCurrentPage(searchParams)).toBe(14);
  });

  test('getCurrentPage() should return 1 if page parameter is not provided', () => {
    const searchParams = new URLSearchParams('?types=fire,electric');
    expect(getCurrentPage(searchParams)).toBe(1);
  });

  test('getCurrentPage() should return 1 if page parameter can not be typecasted no integer', () => {
    const searchParams = new URLSearchParams('?page=second');
    expect(getCurrentPage(searchParams)).toBe(1);
  });
});