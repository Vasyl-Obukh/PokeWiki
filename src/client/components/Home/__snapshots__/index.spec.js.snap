// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> tests Should render correctly if searchParams prop is not provided 1`] = `
<div
  className="sc-bdVaJa lbcdeK"
>
  <main
    className="sc-bwzfXH dZFoTS"
  />
  <aside
    className="sc-kAzzGY ivlbmQ"
  >
    <form
      className="sc-jTzLTM bgkoDW"
      onSubmit={[Function]}
    >
      <h2>
        Filter by:
      </h2>
      <h3>
        Types
      </h3>
      <div
        className="sc-fjdhpX jtggXG"
      >
        <div>
          <input
            checked={false}
            id="normal"
            name="types.normal"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="normal"
          >
            normal
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="fighting"
            name="types.fighting"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="fighting"
          >
            fighting
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="flying"
            name="types.flying"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="flying"
          >
            flying
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="poison"
            name="types.poison"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="poison"
          >
            poison
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="ground"
            name="types.ground"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="ground"
          >
            ground
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="rock"
            name="types.rock"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="rock"
          >
            rock
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="bug"
            name="types.bug"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="bug"
          >
            bug
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="ghost"
            name="types.ghost"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="ghost"
          >
            ghost
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="steel"
            name="types.steel"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="steel"
          >
            steel
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="fire"
            name="types.fire"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="fire"
          >
            fire
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="water"
            name="types.water"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="water"
          >
            water
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="grass"
            name="types.grass"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="grass"
          >
            grass
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="electric"
            name="types.electric"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="electric"
          >
            electric
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="psychic"
            name="types.psychic"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="psychic"
          >
            psychic
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="ice"
            name="types.ice"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="ice"
          >
            ice
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="dragon"
            name="types.dragon"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="dragon"
          >
            dragon
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="fairy"
            name="types.fairy"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="fairy"
          >
            fairy
          </label>
        </div>
      </div>
      <h3>
        Evolution levels
      </h3>
      <div
        className="sc-fjdhpX jtggXG"
      >
        <div>
          <input
            checked={false}
            id={1}
            name="evoLevels.1"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor={1}
          >
            1
          </label>
        </div>
        <div>
          <input
            checked={false}
            id={2}
            name="evoLevels.2"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor={2}
          >
            2
          </label>
        </div>
        <div>
          <input
            checked={false}
            id={3}
            name="evoLevels.3"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor={3}
          >
            3
          </label>
        </div>
        <div>
          <input
            checked={false}
            id={4}
            name="evoLevels.4"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor={4}
          >
            4
          </label>
        </div>
      </div>
      <button
        className="sc-cSHVUG jLBIff"
        type="submit"
      >
        filter
      </button>
    </form>
  </aside>
</div>
`;

exports[`<Home /> tests Should render with pagination 1`] = `
<div
  className="sc-bdVaJa lbcdeK"
>
  <main
    className="sc-bwzfXH dZFoTS"
  />
  <aside
    className="sc-kAzzGY ivlbmQ"
  >
    <form
      className="sc-jTzLTM bgkoDW"
      onSubmit={[Function]}
    >
      <h2>
        Filter by:
      </h2>
      <h3>
        Types
      </h3>
      <div
        className="sc-fjdhpX jtggXG"
      >
        <div>
          <input
            checked={false}
            id="normal"
            name="types.normal"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="normal"
          >
            normal
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="fighting"
            name="types.fighting"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="fighting"
          >
            fighting
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="flying"
            name="types.flying"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="flying"
          >
            flying
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="poison"
            name="types.poison"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="poison"
          >
            poison
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="ground"
            name="types.ground"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="ground"
          >
            ground
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="rock"
            name="types.rock"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="rock"
          >
            rock
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="bug"
            name="types.bug"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="bug"
          >
            bug
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="ghost"
            name="types.ghost"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="ghost"
          >
            ghost
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="steel"
            name="types.steel"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="steel"
          >
            steel
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="fire"
            name="types.fire"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="fire"
          >
            fire
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="water"
            name="types.water"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="water"
          >
            water
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="grass"
            name="types.grass"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="grass"
          >
            grass
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="electric"
            name="types.electric"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="electric"
          >
            electric
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="psychic"
            name="types.psychic"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="psychic"
          >
            psychic
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="ice"
            name="types.ice"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="ice"
          >
            ice
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="dragon"
            name="types.dragon"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="dragon"
          >
            dragon
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="fairy"
            name="types.fairy"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="fairy"
          >
            fairy
          </label>
        </div>
      </div>
      <h3>
        Evolution levels
      </h3>
      <div
        className="sc-fjdhpX jtggXG"
      >
        <div>
          <input
            checked={false}
            id={1}
            name="evoLevels.1"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor={1}
          >
            1
          </label>
        </div>
        <div>
          <input
            checked={false}
            id={2}
            name="evoLevels.2"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor={2}
          >
            2
          </label>
        </div>
        <div>
          <input
            checked={false}
            id={3}
            name="evoLevels.3"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor={3}
          >
            3
          </label>
        </div>
        <div>
          <input
            checked={false}
            id={4}
            name="evoLevels.4"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor={4}
          >
            4
          </label>
        </div>
      </div>
      <button
        className="sc-cSHVUG jLBIff"
        type="submit"
      >
        filter
      </button>
    </form>
  </aside>
</div>
`;

exports[`<Home /> tests Should render without pagination 1`] = `
<div
  className="sc-bdVaJa lbcdeK"
>
  <main
    className="sc-bwzfXH dZFoTS"
  />
  <aside
    className="sc-kAzzGY ivlbmQ"
  >
    <form
      className="sc-jTzLTM bgkoDW"
      onSubmit={[Function]}
    >
      <h2>
        Filter by:
      </h2>
      <h3>
        Types
      </h3>
      <div
        className="sc-fjdhpX jtggXG"
      >
        <div>
          <input
            checked={false}
            id="normal"
            name="types.normal"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="normal"
          >
            normal
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="fighting"
            name="types.fighting"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="fighting"
          >
            fighting
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="flying"
            name="types.flying"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="flying"
          >
            flying
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="poison"
            name="types.poison"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="poison"
          >
            poison
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="ground"
            name="types.ground"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="ground"
          >
            ground
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="rock"
            name="types.rock"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="rock"
          >
            rock
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="bug"
            name="types.bug"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="bug"
          >
            bug
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="ghost"
            name="types.ghost"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="ghost"
          >
            ghost
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="steel"
            name="types.steel"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="steel"
          >
            steel
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="fire"
            name="types.fire"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="fire"
          >
            fire
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="water"
            name="types.water"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="water"
          >
            water
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="grass"
            name="types.grass"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="grass"
          >
            grass
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="electric"
            name="types.electric"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="electric"
          >
            electric
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="psychic"
            name="types.psychic"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="psychic"
          >
            psychic
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="ice"
            name="types.ice"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="ice"
          >
            ice
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="dragon"
            name="types.dragon"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="dragon"
          >
            dragon
          </label>
        </div>
        <div>
          <input
            checked={false}
            id="fairy"
            name="types.fairy"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor="fairy"
          >
            fairy
          </label>
        </div>
      </div>
      <h3>
        Evolution levels
      </h3>
      <div
        className="sc-fjdhpX jtggXG"
      >
        <div>
          <input
            checked={false}
            id={1}
            name="evoLevels.1"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor={1}
          >
            1
          </label>
        </div>
        <div>
          <input
            checked={false}
            id={2}
            name="evoLevels.2"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor={2}
          >
            2
          </label>
        </div>
        <div>
          <input
            checked={false}
            id={3}
            name="evoLevels.3"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor={3}
          >
            3
          </label>
        </div>
        <div>
          <input
            checked={false}
            id={4}
            name="evoLevels.4"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="checkbox"
            value=""
          />
          <label
            className="sc-jzJRlG kYjAMg"
            htmlFor={4}
          >
            4
          </label>
        </div>
      </div>
      <button
        className="sc-cSHVUG jLBIff"
        type="submit"
      >
        filter
      </button>
    </form>
  </aside>
</div>
`;
